rust/tools/authorization-logic/src/parsing/astconstructionvisitor.rs

```diff
@@ -38,8 +38,22 @@ pub fn parse_program(prog_text: &str) -> AstProgram {
     construct_program(&parse_result)
 }
 
+// The authorization logic frontend supports newlines
+// in the names of constants, but souffle does not. As
+// an easy way to support this, we delete newlines from
+// constant names. Since multiline literals/constants
+// are represented with triple quotes, we replace
+// triple quotes with single here.
+fn sanitize_id(name: String) -> String {
+    name.replace("\n", "").replace("\"\"\"", "\"")
+}
+
+fn sanitize_filepath(name: String) -> String {
+    name.replace("\"", "")
+}
```

Comment:
So in the syntax for auth logic prototyped here file paths are used: 1) as paths for claims being imported/exported, 2) to give paths for public/private cryptographic keys. For either case, the file path as written in the auth logic frontend can either be wrapped in quotes or not. I think the tests write file paths both with and without quotes. The collisions are fine since IIUC file paths usually do not allow quotes in them, so we have no need to be able to refer to file paths that actually have quotes.

I think we can/should iterate on the actual syntax we want to keep, but since we are deprecating this version of auth logic, it is probably best if we turn that nob after we have migrated the transparent release work to use the C++ implementation rather than this one.